src/mame/drivers/atari400.cpp

```diff
@@ -2344,9 +2370,11 @@ void a400_state::a5200(machine_config &config)
 	ATARI_ANTIC(config, m_antic, 0);
 	m_antic->set_gtia_tag(m_gtia);
 
-	m_pia->readpa_handler().set_constant(0); // FIXME: is there anything connected here
-	m_pia->readpb_handler().set_constant(0); // FIXME: is there anything connected here
-	m_pia->cb2_handler().set_nop(); // FIXME: is there anything connected here
+	// PIA chip is not present
+	m_pia->readpa_handler().set_constant(0);
+	m_pia->readpb_handler().set_constant(0);
+	m_pia->cb2_handler().set_nop();
+	config.device_remove("sio");
```

Comment:
If the PIA and SIO really aren’t present, you’re better off not adding them in the base machine configuration function.  Using `device_remove` and having vestigial devices hanging around makes life harder.

Can you make it so these things are only added for systems that actually have them present?